bin/filter_header.pl

```diff
@@ -0,0 +1,221 @@
+                $sm = $1;
+                $l[$i] = q[SM:] . $sample;
+                _compare_tag_values(q[SM],$sm, $sample, $rpt)
+            }elsif($l[$i] =~ /^LB:(.*)$/xms){
+                $lb = $1;
+                $l[$i] = q[LB:] . $library;
+                _compare_tag_values(q[LB], $lb, $library, $rpt)
+            }elsif($l[$i] =~ /^DS:(.*)$/xms){
+                $ds = $1;
+                $l[$i] = q[DS:] . $study;
+                _compare_tag_values(q[DS], $ds, $study, $rpt)
+            }
+        }
+        print join(qq[\t], @l) ."\n";
+    }else{
+        print "$_\n";
+    }
+}
+
+sub _get_limsm {
+    my ($lims) = @_;
+    my(@samples,@studies,%s,);
+    foreach my $plex ($lims->children) {
+        next if $plex->is_phix_spike;
+        my ($sample_name,$library_id,$study) = _get_limsi($plex);
+        push @samples, $sample_name;
+        push @studies, $study if ! defined $s{$study};
+        $s{$study}++;
+    }
+    my $sample_list = join ',', @samples;
+    my $study_list  = join ',', @studies;
+    return($sample_list, 'unknown', q[Study ]. $study_list);
+}
+
+sub _get_limsi {
+    my ($lims) = @_;
+    my $sample_name       = _check_tag_value($lims->sample_publishable_name());
+    my $library_id        = _check_tag_value($lims->library_id());
+    my $study_name        = _check_tag_value($lims->study_publishable_name());
+    my $study_description = _check_tag_value($lims->study_description());
+    if($lims->is_phix_spike){
+        $study_description = 'SPIKED_CONTROL'
+    }
+    return($sample_name, $library_id, $study_name. q[: ].$study_description);
+}
+
+sub _check_tag_value {
+    my ($tag_value) = @_;
+    $tag_value =~ s/\n/\ /gmxs;
+    $tag_value =~ s/\t/\ /gmxs;
+    return $tag_value;
+}
+
+sub _compare_tag_values {
+    my ($tag, $file_val, $lims_val, $rpt) = @_;
+    if($file_val ne $lims_val){
+        my $msg = qq[WARNING: LIMS data for: '$rpt', has changed: ];
+        $msg .= qq[$tag old: '$file_val', ];
+        $msg .= qq[$tag new: '$lims_val'.];
+        carp q[$msg];
+    }
+}
+
+__END__
+
+=head1 NAME
+
+filter_header.pl
+
+=head1 VERSION
+
+=head1 USAGE
+
+samtools view -H 6551_1#1.cram | filter_header.pl -rpt 6551_1#1
+
+=head1 CONFIGURATION
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+Reads in a BAM/CRAM file header and prints out an updated version of
+the SM, LB and SM tags in the @RG section. Inteded to be used for
+re-headering of such files.
+
+Uses the value of the -rpt argument to obtain meta data from the
+LIMS to update the values of the SM, LB and DS tags in the @RG
+section. At the same time, it compares what is already there (or not)
+and reports it by printing the results in STDERR. For the value of
+-rpt to be valid it has to be of the form:
+
+run_position[#tag[_split]|_split].
+
+The input and output is the same: a BAM/CRAM file header in human-
+readable format (e.g. using samtools view -H).
+
+=head1 SUBROUTINES/METHODS
+
+=over
+
+=item  _get_limsm
```

Comment:
It's unusual to PerlDoc functions/methods marked as "private" i.e. with a ```_``` prefix - lose such PerlDoc and move the text to comments by the methods?